gcp/iam/workload_identity_federation/README.md

```diff
@@ -0,0 +1,227 @@
+| `providers.owner`                  |    string    |  false   | If using a preconfigured `providers.oidc.issuer` this references the "owner" of the issuer, i.e. the organization or username.                    |                                                            none                                                             |
+| `providers.workspace_uuid`         |    string    |  false   | If `providers.oidc.issuer` is `bitbucket-pipelines`, this references the workspace uuid with the format: `{XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX}` |                                                            none                                                             |
+| `providers.oidc`                   |     map      |  false   | The configuration for an OIDC provider (Either this OR `aws` block can be set)                                                                    |                                                            none                                                             |
+| `providers.oidc.issuer`            |    string    |   true   | The preconfigured template to use, or the OIDC issuer uri                                                                                         |                                                            none                                                             |
+| `providers.oidc.allowed_audiences` | list(string) |  false   | Acceptable values for the `aud` field                                                                                                             | `"https://iam.googleapis.com/projects/project-number/locations/global/workloadIdentityPools/pool-id/providers/provider-id"` |
+| `providers.aws`                    |     map      |  false   | The configuration for an AWS provider (Either this OR `oidc` block can be set)                                                                    |                                                            none                                                             |
+| `providersaws.account_id`          |     map      |   true   | The id of the client aws account                                                                                                                  |                                                            none                                                             |
+
+#### Example
+```terraform
+module workload_identity_pool {
+  source                 = "github.com/mesoform/terraform-infrastructure-modules//gcp/iam/workload_identity_federation"
+  project_id             = "project_id"
+  workload_identity_pool = {
+    pool_id   = "cicd"
+    display_name = "CI/CD pool"
+    providers = {
+      github = {
```

Comment:
I assume that this can be anything and allows the ability to have more than one Github provider configured? If so, make the description clear on that (e.g. github-for-something)